src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -1,103 +1,1003 @@
+                            sQuestSongPlayedOcarinaButtonsNum++;
+                        }
+                    }
+                }
+
+                // Draw the buttons colored as the player plays the song
+                j = i + 32;
+                var_s3 = 0;
+                for (; var_s3 < 8; var_s3++, j += 4) {
+                    if (sQuestSongPlayedOcarinaButtons[var_s3] == OCARINA_BTN_INVALID) {
+                        continue;
+                    }
+
+                    if (sQuestSongPlayedOcarinaButtonsAlpha[var_s3] != 255) {
+                        sQuestSongPlayedOcarinaButtonsAlpha[var_s3] += 50;
+                        if (sQuestSongPlayedOcarinaButtonsAlpha[var_s3] >= 255) {
+                            sQuestSongPlayedOcarinaButtonsAlpha[var_s3] = 255;
+                        }
+                    }
+
+                    pauseCtx->questVtx[j + 0].v.ob[1] = pauseCtx->questVtx[j + 1].v.ob[1] =
+                        pauseCtx->ocarinaButtonsY[sQuestSongPlayedOcarinaButtons[var_s3]];
 
-s16 D_8082AFC4[] = {
+                    pauseCtx->questVtx[j + 2].v.ob[1] = pauseCtx->questVtx[j + 3].v.ob[1] =
+                        pauseCtx->questVtx[j + 0].v.ob[1] - 12;
+
+                    gDPPipeSync(POLY_OPA_DISP++);
+
+                    if (sQuestSongPlayedOcarinaButtons[var_s3] == OCARINA_BTN_A) {
+                        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 80, 150, 255,
+                                        sQuestSongPlayedOcarinaButtonsAlpha[var_s3]);
+                    } else {
+                        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 50,
+                                        sQuestSongPlayedOcarinaButtonsAlpha[var_s3]);
+                    }
+
+                    gSPVertex(POLY_OPA_DISP++, &pauseCtx->questVtx[j], 4, 0);
+
+                    POLY_OPA_DISP = func_8010DC58(
+                        POLY_OPA_DISP, sOcarinaButtonTextures[sQuestSongPlayedOcarinaButtons[var_s3]], 16, 16, 0);
+                }
+
+                if (pauseCtx->unk_200 == 4) {
+                    for (var_s3 = 0; var_s3 < 8; var_s3++) {
+                        sQuestSongPlayedOcarinaButtons[var_s3] = OCARINA_BTN_INVALID;
+                        sQuestSongPlayedOcarinaButtonsAlpha[var_s3] = 0;
+                    }
+
+                    sQuestSongPlayedOcarinaButtonsNum = 0;
+                    AudioOcarina_SetInstrument(OCARINA_INSTRUMENT_DEFAULT);
+                    AudioOcarina_StartDefault((1 << pauseCtx->ocarinaSongIndex) | 0x80000000);
+                    pauseCtx->ocarinaStaff = AudioOcarina_GetPlaybackStaff();
+                    pauseCtx->ocarinaStaff->pos = 0;
+                    pauseCtx->ocarinaStaff->state = 0xFE;
+                    pauseCtx->unk_200 = 5;
+                }
+            }
+        }
+    }
+
+    // Draw Skull Token Count
+    // QUEST_SKULL_TOKEN never properly set, see `Item_Give`,
```

Comment:
```suggestion
    // QUEST_SKULL_TOKEN never properly set, see Item_Give(),
```
Think this gets Doxygen to link it automatically.